paths_local.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+// +build local
+
+package main
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+)
+
+func getRunningBinaryPath() string {
+	return filepath.Dir(os.Args[0])
+}
+
+func getDataDirPath() string {
+	return path.Join(getRunningBinaryPath(), "support")
+}
+
+func getStateDirPath() string {
+	return getRunningBinaryPath()
+}
+
+func getConfDirPath() string {
+	return getRunningBinaryPath()
+}
+
+func getBinDirPath() string {
+	return path.Join(getRunningBinaryPath(), "support")
```

Comment:
On Fri, Sep 9, 2016 at 8:47 AM, Kristian Amlie <notifications@github.com>
wrote:

> In paths_local.go
> <https://github.com/mendersoftware/mender/pull/108#discussion_r78135451>:
>
> > +}
> > +
> > +func getDataDirPath() string {
> > +	return path.Join(getRunningBinaryPath(), "support")
> > +}
> > +
> > +func getStateDirPath() string {
> > +	return getRunningBinaryPath()
> > +}
> > +
> > +func getConfDirPath() string {
> > +	return getRunningBinaryPath()
> > +}
> > +
> > +func getBinDirPath() string {
> > +	return path.Join(getRunningBinaryPath(), "support")
>
> Is it right to append "support" to the end here? Won't the binary end up
> in the mender root directory when you run go build?
>
It will. We also keep sample scripts/tools in $(srcdir)/support, so that's
where `getBinDirPath()` will point to. I run mender right from the source
directory, not sure what others do. Feel free to suggest alternatives.
Perhaps `os.Getwd()`?